velero-plugin-for-aws/object_store_test.go

```diff
@@ -152,3 +165,91 @@ func TestValidChecksumAlg(t *testing.T) {
 		})
 	}
 }
+
+func TestObjectStore_PutObject(t *testing.T) {
+	type args struct {
+		bucket string
+		key    string
+		body   io.Reader
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+		wantLogs []string
+	}{
+		{
+			name:    "PutObject produces logs",
+			wantLogs: []string{
+				`Request Signature:\n---[ CANONICAL STRING  ]-----------------------------\nPUT\n/mybucket/mykey\nx-id=PutObject\naccept-encoding:identity\namz-sdk-invocation-id:`,
+				`x-amz-security-token:SESSION\n\naccept-encoding;amz-sdk-invocation-id;amz-sdk-request;content-length;content-type;host;x-amz-content-sha256;x-amz-date;x-amz-security-token`,
+				`Amz-Sdk-Request: attempt=1; max=3\r\nAuthorization: AWS4-HMAC-SHA256 Credential=KEY`,
```

Comment:
`Credential=KEY` I think this is just Access Key. Hopefully Secret Access Key won't be outputted.

Reply:
@kaovilai did you confirm that it's just Access Key?

Reply:
afaict, I am not seeing `SECRET` in the logs, which would indicate secret access key..

Reply:
@Wayne-H-Ha can you confirm if the logs did not show secretAccessKey?

Reply:
I see my access key id in the log:
```
Authorization: AWS4-HMAC-SHA256 Credential=MY_access_key_id_0123456789abcde/20240703/eu-de/s3/aws4_request, 
SignedHeaders=accept-encoding;amz-sdk-invocation-id;amz-sdk-request;content-length;content-type;host;x-amz-content-sha256;x-amz-date, 
Signature=0123456789abcdefxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx\r\nContent-Type: application/octet-stream\r\nX-Amz-Content-Sha256:
```

Reply:
Not as concerned with accessKey. Mostly secretKey. If that's not in the logs we should be good here.

Reply:
Correct, secret_access_key is not found in the log